cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -340,3 +379,229 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration, volumes []v1.Vo
 func buildStaticManifestFilepath(name string) string {
 	return path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "manifests", name+".yaml")
 }
+
+func apiServerPKISecretVolumes() []v1.Volume {
+	return []v1.Volume{
+		v1.Volume{
+			Name: "k8s-pki",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName,
+				},
+			},
+		},
+		v1.Volume{
+			Name: "k8s-pki",
```

Comment:
They definitely should... working on that now. The previous version had a big monolithic secret and then used `Items` to choose what to mount.